packages/fxa-auth-server/lib/oauthdb/index.js

```diff
@@ -99,7 +101,19 @@ module.exports = (log, config) => {
 
     async grantTokensFromAuthorizationCode(oauthParams) {
       try {
-        return await api.grantTokensFromAuthorizationCode(oauthParams);
+        const codeResponse = await api.grantTokensFromAuthorizationCode(
+          oauthParams
+        );
+        const scopeSet = ScopeSet.fromString(codeResponse.scope);
+        if (scopeSet.contains(SESSION_TOKEN_OAUTH_SCOPE)) {
+          // at this point the auth code request has
+          // validated that the session scope is allowed for this client
+          codeResponse.session_token = await db.createSessionToken({
+            uid: codeResponse.uid,
+          });
+        }
+
+        return codeResponse;
```

Comment:
`session_token` here is 

> {
  "data": "a5f118d0ecbbeaa2aaf45d1c5bb3d9234d50ada7aa8a93f1e1a84e8735bb4935",
  "id": "090d066d5efdb01c10e222b7abbcb7bc12d4769ba90ff4fa38aac8ffd210f89a",
  "authKey": "00a2ca8b8ec18be614873f61d5330724c284953672908cfb08b3bf9636215a0c",
  "bundleKey": "5f877ab616474f119db6be98502af6289b7ccda5f5a74852437849fab9d46977",
  "uid": "26f9b9198abd401c817e8b752eaaac4f",
  "lifetime": 2419200000,
  "createdAt": 1567967475605,
  "email": null,
  "emailCode": null,
  "emailVerified": false,
  "authAt": 0,
  "locale": null,
  "mustVerify": false,
  "tokenVerificationId": null,
  "tokenVerified": true,
  "tokenVerificationCode": null,
  "tokenVerificationCodeExpiresAt": null,
  "verificationMethod": null,
  "verifiedAt": null
}

atm